Android/app/src/main/java/com/example/hermes/CarControl.java

```diff
@@ -28,54 +30,25 @@
 
 // Adapted from https://github.com/DIT112-V21/smartcar-mqtt-controller.git for mqtt connection
 
-public class CarControl extends AppCompatActivity {
+public class CarControl extends AbstractSteering {
 
-
-    private AppBarConfiguration appBarConfiguration;
     private ActivityCarControlBinding binding;
-    private MqttClient mqttClient;
-    private  MqttClient mqttClientExternal;
-    private static final  String EXTERNAL_MQTT_BROKER = "aerostun.dev";
-    private static final String LOCALHOST = "10.0.2.2";
-    private static final String MQTT_SERVER = "tcp://" + LOCALHOST + ":1883";
-    private static final String TAG = "MqttController";
-    private ImageView camera;
-    private static final int QOS = 1;
-    private boolean isConnected = false;
-    private static final int IMAGE_WIDTH = 320;
-    private static final int IMAGE_HEIGHT = 240;
-    private static final String THROTTLE_CONTROL = "/smartcar/control/throttle";
-    private static final String STEERING_CONTROL = "/smartcar/control/steering";
-
     private static final int MOVEMENT_SPEED = 70;
     private static final int IDLE_SPEED = 0;
     private static final int STRAIGHT_ANGLE = 0;
     private static final int STEERING_ANGLE = 50;
+    private  ImageView camera;
 
+    @SuppressLint("ClickableViewAccessibility")
     @Override
     protected void onCreate(Bundle savedInstanceState) {
-
+        setContentView(R.layout.activity_car_control);
         super.onCreate(savedInstanceState);
-
+        initialiseMqttClient(getApplicationContext());
         binding = ActivityCarControlBinding.inflate(getLayoutInflater());
         setContentView(binding.getRoot());
-
-       // setSupportActionBar(binding.toolbar);
-
-       // NavController navController = Navigation.findNavController(this, R.id.nav_host_fragment_content_car_control);
-        //appBarConfiguration = new AppBarConfiguration.Builder(navController.getGraph()).build();
-        //NavigationUI.setupActionBarWithNavController(this, navController, appBarConfiguration);
-
-      /* binding.fab.setOnClickListener(new View.OnClickListener() {
```

Comment:
Since the code is commented, it could be deleted for having a better looking code format.